***************
*** 1227,1232 ****
--- 1227,1233 ----
  {
  	int i;
  	const struct fb_videomode *default_vmode = NULL;
+ 	struct fb_videomode the_vmode;
  	int result = 0;
  	char *edid;
  	int tries = 3;
***************
*** 1338,1344 ****
  	/* If we have good mode and no active clients*/
  	if ((default_vmode != NULL) && (dev->fb_count == 0)) {
  
! 		fb_videomode_to_var(&info->var, default_vmode);
  		dlfb_var_color_format(&info->var);
  
  		/*
--- 1339,1348 ----
  	/* If we have good mode and no active clients*/
  	if ((default_vmode != NULL) && (dev->fb_count == 0)) {
  
! 		memcpy(&the_vmode, default_vmode, sizeof(struct fb_videomode));
! 		the_vmode.xres = 1200;
! 		the_vmode.yres = 824;
! 		fb_videomode_to_var(&info->var, &the_vmode);
  		dlfb_var_color_format(&info->var);
  
  		/*
